### ISSUE TITLE
computational complexity

### ISSUE THREAD

**Issue by yejing0** [facebookresearch/segment-anything#630]
Very excellent work, but we would like to further evaluate the efficacy of the model.
How do we test the FLOPs of the model?

**Comment by BeliverK** [comment]
> 我不确定是否有一种完美的方法来计算 FLOP。一种选择是使用一些软件来估计模型定义中的计数。四处搜索，看起来 facebookresearch 有另一个包含此功能的存储库 （[fvcore](https://github.com/facebookresearch/fvcore/blob/main/docs/flop_count.md)），您可能想尝试一下。
> 
> 另一种选择是从模型定义中手动估计计数，这说起来容易做起来难！话虽如此，SAM 模型的大部分是基于转换器的，因此，如果您估计[图像编码器](https://github.com/facebookresearch/segment-anything/blob/6fdee8f2727f4506cfbbe553e23b895e27956588/segment_anything/modeling/image_encoder.py#L166C1-L182C17)和[掩码解码器](https://github.com/facebookresearch/segment-anything/blob/6fdee8f2727f4506cfbbe553e23b895e27956588/segment_anything/modeling/transformer.py#L62C1-L106C29)的单个转换器块的 FLOP 计数，则只需乘以模型每个组件中的转换器块数量（图像编码器和掩码器大小配置都可以在[build_sam.py](https://github.com/facebookresearch/segment-anything/blob/main/segment_anything/build_sam.py)脚本中找到），就可以获得总 FLOP 计数的近似值。
Very excellent work！
I would like to ask, have you tried to use fvcore to calculate FLOPs, what parts are included in your calculation, I did not succeed in the calculation, I would like to ask if you have a calculation code

**Comment by BeliverK** [comment]
from calflops import calculate_flops
from torchvision import models
import requests
from PIL import Image
from transformers import SamModel, SamProcessor
import torch 
device = "cuda" if torch.cuda.is_available() else "cpu"
model = SamModel.from_pretrained("./sam-vitb")
processor = SamProcessor.from_pretrained("./sam-vitb")

img_pth = "./car.png"
raw_image = Image.open(img_pth).convert("RGB")
input_points = [[[450, 600,2000,1000],
                        ]]
inputs = processor(raw_image, input_boxes=input_points, return_tensors="pt")
outputs = model(**inputs)
masks = processor.image_processor.post_process_masks(outputs.pred_masks.cpu(), inputs["original_sizes"].cpu(), inputs["reshaped_input_sizes"].cpu())
flops, macs, params = calculate_flops(model=model,
                                      kwargs=inputs,   
                                      print_results=True)
print("Salesforce/blip2-opt-2.7b FLOPs:%s   MACs:%s   Params:%s \n" % (flops, macs, params))
I used calflops to test lSAM-Base and SAM-Large, the code is as shown in the image, I didn't test the FLOPs of SAM-H due to memory limitations。But I now want to test FLOPs for MobileSAM and it hasn't worked out yet

**Comment by eleveneee** [comment]
> from calflops import calculate_flops from torchvision import models import requests from PIL import Image from transformers import SamModel, SamProcessor import torch device = "cuda" if torch.cuda.is_available() else "cpu" model = SamModel.from_pretrained("./sam-vitb") processor = SamProcessor.from_pretrained("./sam-vitb")
> 
> img_pth = "./car.png" raw_image = Image.open(img_pth).convert("RGB") input_points = [[[450, 600,2000,1000], ]] inputs = processor(raw_image, input_boxes=input_points, return_tensors="pt") outputs = model(**inputs) masks = processor.image_processor.post_process_masks(outputs.pred_masks.cpu(), inputs["original_sizes"].cpu(), inputs["reshaped_input_sizes"].cpu()) flops, macs, params = calculate_flops(model=model, kwargs=inputs, print_results=True) print("Salesforce/blip2-opt-2.7b FLOPs:%s MACs:%s Params:%s \n" % (flops, macs, params)) I used calflops to test lSAM-Base and SAM-Large, the code is as shown in the image, I didn't test the FLOPs of SAM-H due to memory limitations。But I now want to test FLOPs for MobileSAM and it hasn't worked out yet

Hello, have you calculated it yet? I also calculated this data, looking forward to your reply
